ddtrace/_worker.py

```diff
@@ -60,7 +63,8 @@ def is_alive(self):
         return self._thread.is_alive()
 
     def join(self, timeout=None):
-        return self._thread.join(timeout)
+        if self.is_alive():
+            return self._thread.join(timeout)
```

Comment:
This is a possible race condition. If the thread exists after `is_alive()` is called, that fails.